### PR TITLE
Move Matrix test out of test_matrices.py

### PR DESCRIPTION
**Pull request by Arpan612**
Move Matrix test out of test_matrices.py
#### References to other Issues or PRs

Related to #18666 

#### Brief description of what is fixed or changed
The functions of Matrix tests were moved out of test_matrices.py to test_reductions.py

Other comments

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by Arpan612** [comment]
Could someone tell me where is the problem?

**Comment by jksuom** [comment]
flake8 has found some issues. See the Travis log https://travis-ci.org/github/sympy/sympy/jobs/666275053?utm_medium=notification&utm_source=github_status

**Comment by Arpan612** [comment]
@jksuom Thanks. Working On it.

File: sympy/matrices/tests/test_reductions.py [hunk 1]
```diff
@@ -1,11 +1,13 @@
...
 from sympy.matrices.common import _MinimalMatrix, _CastableMatrix
 from sympy.matrices.matrices import MatrixReductions
 from sympy.testing.pytest import raises
-from sympy.matrices import Matrix
+from sympy.matrices import Matrix, zeros
 from sympy.core.symbol import Symbol
 from sympy.core.numbers import Rational
 from sympy.functions.elementary.miscellaneous import sqrt
 from sympy.simplify.simplify import simplify
+from sympy.abc import a, b, c, d, e, x
```
Comment (mohitacecode) [inline sympy/matrices/tests/test_reductions.py]: I did'nt see any use of this import you can remove this line .

File: sympy/matrices/tests/test_reductions.py [hunk 2]
```diff
@@ -1,4 +1,4 @@
-from sympy import symbols, simplify, I
+from sympy import symbols, I
```
Comment (smitgajjar) [inline sympy/matrices/tests/test_reductions.py]: Remove comma after symbols too

Comment (Arpan612) [reply sympy/matrices/tests/test_reductions.py]: I have already done that, thanks though.

Comment (smitgajjar) [reply sympy/matrices/tests/test_reductions.py]: Also, error is on line 9 too as sugggested by flake8, remove the simplify import

File: sympy/matrices/tests/test_reductions.py [hunk 3]
```diff
@@ -275,3 +276,70 @@ def test_rref():
...
+    assert(E[5, :] == Matrix([[0, 3, 0, 14]]))
+    assert(F[5, :] == Matrix([[16, 30, 0, 12]]))
+    # Tests for row/col out of range
+    raises(ValueError, lambda: C.elementary_row_op('n<->m', row1=2, row2=6))
+    raises(ValueError, lambda: C.elementary_row_op('n->kn', row=7, k=2))
+    raises(ValueError, lambda: C.elementary_row_op('n->n+km', row1=-1, row2=5, k=2))
+
+def test_rank():
+    from sympy.abc import x
+    m = Matrix([[1, 2], [x, 1 - 1/x]])
```
Comment (Smit-create) [inline sympy/matrices/tests/test_reductions.py]: Move this import at the top of the file.

File: sympy/matrices/tests/test_reductions.py (hunk 2 above)
Comment (Arpan612) [reply sympy/matrices/tests/test_reductions.py]: Have done that, please check.

File: sympy/matrices/tests/test_reductions.py (hunk 3 above)
Comment (Arpan612) [reply sympy/matrices/tests/test_reductions.py]: Have done that, please check.

File: sympy/matrices/tests/test_reductions.py [hunk 4]
```diff
@@ -1,11 +1,13 @@
...
 from sympy.matrices.common import _MinimalMatrix, _CastableMatrix
 from sympy.matrices.matrices import MatrixReductions
 from sympy.testing.pytest import raises
-from sympy.matrices import Matrix
+from sympy.matrices import Matrix, zeros
 from sympy.core.symbol import Symbol
 from sympy.core.numbers import Rational
 from sympy.functions.elementary.miscellaneous import sqrt
-from sympy.simplify.simplify import simplify
+#from sympy.simplify.simplify import simplify
```
Comment (mohitacecode) [inline sympy/matrices/tests/test_reductions.py]: don't comment it is better to remove if it is not used.

Comment (mohitacecode) [reply sympy/matrices/tests/test_reductions.py]: from the flake8 error It seems that it is required so please uncomment it.

Comment (Arpan612) [reply sympy/matrices/tests/test_reductions.py]: Sure. Working on it.

**Comment by Sc0rpi0n101** [comment]
@Arpan612 Please make sure you squash all your commits into one with a proper description of what was changed in the commit after you've finalized the changes.

**Comment by Arpan612** [comment]
@sylee957 Sir, could you please review the changes?

**Comment by Arpan612** [comment]
> @Arpan612 Please make sure you squash all your commits into one with a proper description of what was changed in the commit after you've finalized the changes.

I tried my best to squash them, but there is a continuous compatibility error coming up. I assure you , the changes are proper.

**Comment by Arpan612** [comment]
@Sc0rpi0n101 I tried to squash it by making a new PR. I am able to squash the last 5 changes but not the ones before it. This problem I think will continue when I send a new PR for a new issue as well. Could you tell me what to do?

**Comment by sylee957** [comment]
I am merging this, but I think that you should not mess up with your master branch
https://github.com/Arpan612/sympy/tree/master
and squash some redundant commits.

**Comment by Arpan612** [comment]
@sylee957 Thank you sir, I will keep it in my mind.

**Pull request merged**